src/Orchard.Cms.Web/Modules/Orchard.Tenants/Controllers/AdminController.cs

```diff
@@ -0,0 +1,369 @@
+            }
+
+            shellSettings.State = TenantState.Running;
+            _orchardHost.UpdateShellSettings(shellSettings);
+
+            return RedirectToAction(nameof(Index));
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> Reload(string id)
+        {
+            if (!await _authorizationService.AuthorizeAsync(User, Permissions.ManageTenants))
+            {
+                return Unauthorized();
+            }
+
+            if (_currentShellSettings.Name != ShellHelper.DefaultShellName)
+            {
+                return Unauthorized();
+            }
+
+            var shellContext = _orchardHost
+                .ListShellContexts()
+                .OrderBy(x => x.Settings.Name)
+                .Where(x => String.Equals(x.Settings.Name, id, StringComparison.OrdinalIgnoreCase))
+                .FirstOrDefault();
+
+            if (shellContext == null)
+            {
+                return NotFound();
+            }
+
+            var shellSettings = shellContext.Settings;
+            _orchardHost.ReloadShellContext(shellSettings);
+
+            return RedirectToAction(nameof(Index));
+        }
+        private void ValidateViewModel(EditTenantViewModel model, bool newTenant)
+        {
+            if (String.IsNullOrWhiteSpace(model.Name))
+            {
+                ModelState.AddModelError(nameof(EditTenantViewModel.Name), S["The tenant name is mandatory."]);
+            }
+
+            var allShells = GetShells();
+
+            if (newTenant && allShells.Any(tenant => String.Equals(tenant.Settings.Name, model.Name, StringComparison.OrdinalIgnoreCase)))
+            {
+                ModelState.AddModelError(nameof(EditTenantViewModel.Name), S["A tenant with the same name already exists.", model.Name]);
+            }
+
+            if (!String.IsNullOrEmpty(model.Name) && !Regex.IsMatch(model.Name, @"^\w+$"))
+            {
+                ModelState.AddModelError(nameof(EditTenantViewModel.Name), S["Invalid tenant name. Must contain characters only and no spaces."]);
+            }
+
+            if (!String.Equals(model.Name, ShellHelper.DefaultShellName, StringComparison.OrdinalIgnoreCase) && string.IsNullOrWhiteSpace(model.RequestUrlHost) && string.IsNullOrWhiteSpace(model.RequestUrlPrefix))
+            {
+                ModelState.AddModelError(nameof(EditTenantViewModel.RequestUrlPrefix), S["Host and url prefix can not be empty at the same time."]);
+            }
+
+            if (!String.IsNullOrWhiteSpace(model.RequestUrlPrefix))
+            {
+                if (model.RequestUrlPrefix.Contains('/'))
+                {
+                    ModelState.AddModelError(nameof(EditTenantViewModel.RequestUrlPrefix), S["The url prefix can not contains more than one segment."]);
+                }
+
+                if (allShells.Any(x => String.Equals(x.Settings.RequestUrlPrefix.Trim(), model.RequestUrlPrefix.Trim(), StringComparison.OrdinalIgnoreCase)))
```

Comment:
Settings.RequestUrlPrefix is null for default site.

Prevent null reference : Settings.RequestUrlPrefix?.Trim()